contracts/examples/ReferenceCosigner.sol

```diff
@@ -0,0 +1,432 @@
+            This porcentage its not a real, will divide for 10000 instead of 100, in example: 1 its equal to 0.01%
+        @return requiredArrears Delta time, the time required to the loan be in default
+        @return expiration The timeStamp in which the signature will be valid
+    */
+    function _decodeCosignerData(
+        bytes _data
+    ) internal pure returns (uint128, uint16, uint64, uint64) {
+        require(_data.length == L_DATA, "Invalid data length");
+        (bytes32 cost, bytes32 coverage, bytes32 requiredArrears, bytes32 expiration) = decode(
+            _data,
+            16,
+            2,
+            8,
+            8
+        );
+
+        return (uint128(cost), uint16(coverage), uint64(requiredArrears), uint64(expiration));
+    }
+
+    function encodeData(
+        uint128 _cost,
+        uint16 _coverage,
+        uint64 _requiredArrears,
+        uint64 _expiration
+    ) external view returns (bytes) {
+        return abi.encodePacked(
+            _cost,
+            _coverage,
+            _requiredArrears,
+            _expiration
+        );
+    }
+
+    function hashDataSignature(
+        ILoanManager _loanManager,
+        uint256 _index,
+        uint128 _cost,
+        uint16 _coverage,
+        uint64 _requiredArrears,
+        uint64 _expiration
+    ) external view returns (bytes32) {
+        return keccak256(
+            abi.encodePacked(
+                address(this),
+                _loanManager,
+                _index,
+                _cost,
+                _coverage,
+                _requiredArrears,
+                _expiration
+            )
+        );
+    }
+}
+
+
+contract Events {
+    event ReadedOracle(
+        address _oracle,
+        uint256 _tokens,
+        uint256 _equivalent
+    );
+
+    event Received(
+        address _operator,
+        address _from,
+        uint256 _id,
+        bytes _data
+    );
+}
+
+
+contract ReferenceCosigner is SimpleDelegable, Cosigner, Helper, Events, IERC721Receiver {
+    using SafeMath for uint256;
+
+    string private infoUrl;
+    Token public rcn;
+
+    struct Liability {
+        uint16 coverage;
+        uint64 requiredArrears;
+    }
+    // loanManager to loan index to liability
+    mapping(address => mapping(uint256 => Liability)) public liabilities;
+
+    constructor(Token _rcn) public {
+        rcn = _rcn;
+    }
+
+    /**
+        @dev Defines a custom logic that determines if a loan is defaulted or not.
+
+        @param _index Index of the loan
+
+        @return true if the loan is considered defaulted
+    */
+    function isDefaulted(
+        ILoanManager _loanManager,
+        uint256 _index
+    ) external view returns (bool) {
+        return _loanManager.getStatus(_index) != STATUS_PAID &&
+            ((uint256(liabilities[_loanManager][_index].requiredArrears)).add(_loanManager.getDueTime(_index)) <= now);
```

Comment:
`using SafeMath for uint64;`

Can be used to avoid uint256 cast

Reply:
done in [0d49f0d](https://github.com/ripio/rcn-network/pull/103/commits/0d49f0d1353a5793e77a6782627c1fd48b188446)